### PR TITLE
[flutter_tools] Clean up build_aot code post assemble migration

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

As part of the migration to assemble, build_aot was made to delegate to assemble before we updated flutter.gradle. Now that this has been done, the build_aot code is not needed and can be deleted. Note: I left the methods mostly in place since I'll need the same logic for iOS once I start it.

### CODE REVIEW

File: packages/flutter_tools/lib/src/aot.dart [hunk 1]
```diff
@@ -196,6 +195,7 @@ class AotBuilder {
     String outputDir,
     bool quiet
   }) async {
+    // This code is currently dead, but will be updated as we move iOS to assemble.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/aot.dart]: This comment should probably link to a github issue.

Comment (zanderso) [reply packages/flutter_tools/lib/src/aot.dart]: Maybe also `assert(false)` or `throw 'Unreachable'`.

File: packages/flutter_tools/lib/src/commands/assemble.dart [hunk 2]
```diff
@@ -40,9 +40,7 @@ const List<Target> _kDefaultTargets = <Target>[
   FastStartAndroidApplication(),
   ProfileAndroidApplication(),
   ReleaseAndroidApplication(),
-  // These are one-off rules for bundle and aot compat
-  ReleaseCopyFlutterAotBundle(),
-  ProfileCopyFlutterAotBundle(),
+  // These are one-off rules for bundle and aot compat.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/assemble.dart]: "This is a" ?

File: packages/flutter_tools/lib/src/aot.dart [hunk 3]
```diff
@@ -175,7 +175,6 @@ class AotBuilder {
       case TargetPlatform.android_arm64:
       case TargetPlatform.android_x86:
       case TargetPlatform.darwin_x64:
-        return true;
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/aot.dart]: The code below the `if(_canUseAssemble() ...)` above in the `AotBuilder` constructor is also dead with this. Maybe include a comment there like the one below if that is intentional.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/aot.dart]: Done

File: packages/flutter_tools/lib/src/aot.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/aot.dart]: Done

File: packages/flutter_tools/lib/src/commands/assemble.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/assemble.dart]: Done

**Review by zanderso — APPROVED** [review]

**Pull request merged**